framework/data/Filter.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\data;
+
+use Yii;
+use yii\base\InvalidParamException;
+use yii\base\Object;
+use yii\db\ActiveRecord;
+use yii\helpers\Json;
+use yii\web\BadRequestHttpException;
+use yii\web\NotAcceptableHttpException;
+use yii\web\Request;
+
+class Filter extends Object
+{
+    /**
+     * @var string the name of the parameter that specifies which attributes to be sorted
+     * in which direction. Defaults to 'filter'.
+     */
+    public $filterParam = 'filter';
+    /**
+     * @var array
+     */
+    public $operators = ['>=', '<=', '>', '<', '!='];
+    /**
+     * @var array
+     */
+    public $fields = ['*'];
+
+    /**
+     * @param string $modelClass
+     * @return array
+     * @throws NotAcceptableHttpException
+     */
+    public function getConditions($modelClass)
+    {
+        $requestFilter = $this->getRequestFilter();
+        $filter = [];
+
+        if (!empty($requestFilter)) {
+            /** @var $model ActiveRecord */
+            $model = new $modelClass;
+
+            foreach ($requestFilter as $field => $value) {
+                list($operator, $field) = $this->prepareField($field);
+
+                if (!$this->isAcceptableField($field) || !$model->hasAttribute($field)) {
+                    throw new NotAcceptableHttpException('Filter by field "' . $field . '" unsupported.');
+                }
+                                
+                if ($operator) {
```

Comment:
I'd suggest to list in `$operators` property all operators and corresponding methods:

```php
public $operators = [
   '>=' => 'buildGteCondition',
   '<=' => 'buildLteCondition', 
   '>' => 'buildGtCondition',
   '<' => 'buildLtCondition',
   '!=' => 'buildNeqCondition'
];
```

```php
if ($operator) {
    $this->buildCondition($operator, $field, $value);
}
```

```php
public function buildCondition($operator, $field, $value)
{
    if (isset($this->opeartors[$operator])) {
        $method = $this->conditionBuilders[$operator];
    } else {
        $method = 'buildSimpleCondition';
    }

    return $this->$method($operator, $field, $value);
}

protected function buildSimpleCondition($operator, $field, $value)
{
    return [$operator, $field, $value];
}

protected function buildNeqCondition($operator, $field, $value) 
{
     if (is_array($value) && !empty($value)) {
         $operator = 'NOT IN';
     }

     return $this->buildSimpleCondition($operator, $field, $value);
}
```